prime-router/src/main/kotlin/azure/EmailEngineFunction.kt

```diff
@@ -319,21 +348,30 @@ class EmailScheduleEngine {
         mail.setTemplateId(template)
 
         var sendgridId: String? = SecretHelper.getSecretService().fetchSecret("SENDGRID_ID")
+        var response: Response = Response()
 
         if (sendgridId !== null) {
 
             val sg: SendGrid = SendGrid(sendgridId)
             val request: Request = Request()
+
             try {
                 request.setMethod(Method.POST)
                 request.setEndpoint("mail/send")
                 request.setBody(mail.build())
-                sg.api(request)
+                response = sg.api(request)
             } catch (ex: IOException) {
                 logger.warning("Can't contact sendgrid")
             }
+            finally {
+                logger.info( "sending to $emails - result ${response.getStatusCode()}" )
+                if( response.getStatusCode() != 200 )
+                    logger.info( "error - ${response.getBody()}")
```

Comment:
Should this be a `logger.error` instead?

I am not familiar with SendGrid's API, and so this question is more for my information: are there cases where other `2xx` codes are returned indicating success or does it only return `200` on success. For instance (cf. https://en.wikipedia.org/wiki/Http_status_code#2xx_success) Would it make more sense to look for any status code inside the `[200, 299]` range (which I know is broader than what HTTP currently uses, but is semantically correct)